src/utils/twoFactor.js

```diff
@@ -188,34 +113,100 @@ export class TwoFactor {
+            request: {
+                receiver_id: receiverId,
+                actions: convertActions(actions, accountId, receiverId)
+            }
+        })));
+        try {
+            await super.signAndSendTransaction(accountId, [
+                functionCall('add_request_and_confirm', args, GAS_2FA, '0')
+            ])
+            await this.sendRequest(requestId)
+        } catch (e) {
+            throw new WalletError('Error creating request')
+        }
+    }
+
+    async signAndSendTransactions(transactions) {
+        for (let { receiverId, actions } of transactions) {
+            await this.signAndSendTransaction(receiverId, actions)
+        }
+    }
 }
 
-const getContract = (account, accountId) => {
+const convertActions = (actions, accountId, receiverId) => actions.map((a) => {
+    const type = a.enum
+    const { gas, publicKey, methodName, args } = a[type]
+    const action = {
+        ...a[type],
+        type: type[0].toUpperCase() + type.substr(1),
+        gas: (gas && gas.toString()) || undefined,
+        public_key: (publicKey && convertPKForContract(publicKey)) || undefined,
+        method_name: methodName,
+        args: (args && Buffer.from(args).toString('base64')) || undefined,
+    }
+    delete action.publicKey
+    delete action.methodName
+    if (action.deposit) {
+        action.deposit = action.amount = action.deposit.toString()
+    }
+    if (action.accessKey) {
+        if (receiverId === accountId && action.accessKey.permission.enum !== 'fullAccess') {
+            action.permission = {
+                receiver_id: accountId,
+                allowance: ALLOWANCE_2FA,
+                method_names: METHOD_NAMES_LAK,
+            }
+        }
+        if (action.accessKey.permission.enum === 'functionCall') {
+            const { receiverId: receiver_id, methodNames: method_names, allowance,  } = action.accessKey.permission.functionCall
+            action.permission = { receiver_id, allowance, method_names }
+        }
+        delete action.accessKey
+    }
+    return action
+})
+
+const getContract = (account) => {
     return new nearApiJs.Contract(account, account.accountId, {
         viewMethods: VIEW_METHODS,
         changeMethods: METHOD_NAMES_LAK,
     });
 }
 
 const deleteUnconfirmedRequests = async (contract) => {
-    const request_ids = await contract.list_request_ids().catch((e) => { console.log(e) })
-    if (!request_ids || request_ids.length === 0) {
-        return
-    }
+    const request_ids = await contract.list_request_ids()
     for (const request_id of request_ids) {
-        try {
-            await contract.delete_request({ request_id })
-        } catch(e) {
-            console.warn(e)
-        }
+        // TODO discuss if need to show these errors; request cooldown prevents deletion and contract panics. Error is pretty useless to console.warn
+        await contract.delete_request({ request_id }).catch((e) => {})
```

Comment:
let's not use `.catch()` in `async` functions, it can be confusing.

> Error is pretty useless to console.warn

what is the typical reason for error that it's useless from your POV?

Reply:
This method is to clean up any requests we CAN delete (those older than 15 minutes and unconfirmed) for the user's account.

If we fail deleting those requests because of panic (under 15 minutes) we are not concerned.